src/interpolate.jl

```diff
@@ -40,6 +40,113 @@ function cell2node(cell_centered_data::AbstractArray{Float64})
+    error("illegal dimension '$slice_axis', supported dimensions are $supported_dims")
+  end
+  slice_axis_dimension = dimensions[slice_axis]
+  other_dimensions = [1, 2, 3][1:end .!= slice_axis_dimension]
+
+  # Extract data shape information
+  n_nodes_in, _, _, n_elements, n_variables = size(unstructured_data)
+
+  # Get node coordinates for DG locations on reference element
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes_in)
+
+  # New unstructured data has one dimension less.
+  # The redundant element ids are removed later.
+  new_unstructured_data = similar(unstructured_data[1, ..])
+
+  # Declare new empty arrays to fill in new coordinates and levels
+  new_coordinates = Array{Float64}(undef, 2, 0)
+  new_levels = Array{eltype(levels)}(undef, 0)
+
+  # Counter for new element ids
+  new_id = 0
+
+  # Save vandermonde matrices in a Dict to prevent redundant generation
+  vandermonde_to_2d = Dict()
+
+  # Limits of domain in slice_axis dimension
+  lower_limit = center_level_0[slice_axis_dimension] - length_level_0 / 2
+  upper_limit = center_level_0[slice_axis_dimension] + length_level_0 / 2
+
+  if slice_axis_intersect < lower_limit || slice_axis_intersect > upper_limit
+    error("slice_axis_intersect $slice_axis_intersect outside of domain")
+  end
+
+  for element_id in 1:n_elements
+    # Distance from center to border of this element (half the length)
+    element_length = length_level_0 / 2^levels[element_id]
+    first_coordinate = coordinates[:, element_id] .- element_length / 2
+    last_coordinate = coordinates[:, element_id] .+ element_length / 2
+
+    # Check if slice plane and current element intersect
+    # The upper limit check is needed because of the > in the first check
+    if (first_coordinate[slice_axis_dimension] <= slice_axis_intersect &&
+          last_coordinate[slice_axis_dimension] > slice_axis_intersect) ||
+        (slice_axis_intersect == upper_limit &&
+          last_coordinate[slice_axis_dimension] == upper_limit)
+      # This element is of interest
+      new_id += 1
+
+      # Add element to new coordinates and levels
+      new_coordinates = hcat(new_coordinates, coordinates[other_dimensions, element_id])
+      push!(new_levels, levels[element_id])
+
+      # Construct vandermonde matrix (or load from Dict if possible)
+      normalized_intersect =
+          (slice_axis_intersect - first_coordinate[slice_axis_dimension]) /
+          element_length * 2 - 1
+
+      if haskey(vandermonde_to_2d, normalized_intersect)
+        vandermonde = vandermonde_to_2d[normalized_intersect]
+      else
+        # Generate vandermonde matrix to interpolate values at nodes_in to one value
+        vandermonde = polynomial_interpolation_matrix(nodes_in, [normalized_intersect])
+        vandermonde_to_2d[normalized_intersect] = vandermonde
+      end
+
+      # 1D interpolation to specified slice plane
+      for i in 1:n_nodes_in
+        for ii in 1:n_nodes_in
+          if slice_axis == :x
```

Comment:
Reverse the order: First check for the slice_axis, then have the loop. This creates slightly more redundant code, but is much faster since you move the `if` condition outside of the kernel.

Reply:
I don't like the idea of having duplicate code.
The best I came up with is this:
```julia
if slice_axis == :x
  unstructured_data = permutedims(unstructured_data, [2, 3, 1, 4, 5])
elseif slice_axis == :y
  unstructured_data = permutedims(unstructured_data, [1, 3, 2, 4, 5])
end

for i in 1:n_nodes_in
  for ii in 1:n_nodes_in
    data = unstructured_data[i, ii, :, element_id, :]

    value = interpolate_nodes(permutedims(data), vandermonde, n_variables)
    new_unstructured_data[i, ii, new_id, :] = value[:, 1]
  end
end
```
I first permuted the dimensions such that the third dimension is always the one to "remove". It's definitely less readable than the original code, but there is no redundant code and no performance issue.

I think this is acceptable with a good comment and better than redundant code. What do you think?

Reply:
You could also just create a function doing the stuff you would duplicate?

Reply:
I don't care either way: Either redundant code (which seems to be the fastest way to do it), or a really good comment that explains what you are doing without me having to look up what `permutedims` does exactly ;-)

Reply:
@erik-f Once you've resolved this and the PR is ready for a review, just re-request a review